packages/contracts/src/utils/protocol/VersionComparisonLib.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+library VersionComparisonLib {
+    /// @notice Equality comparator for two semantic version numbers.
+    /// @param lhs The left-hand side semantic version number.
+    /// @param lhs The right-hand side semantic version number.
+    /// @return Whether the two numbers are equal or not.
+    function eq(uint8[3] memory lhs, uint8[3] memory rhs) internal pure returns (bool) {
+        if (lhs[0] != rhs[0]) return false;
+
+        if (lhs[1] != rhs[1]) return false;
+
+        if (lhs[2] != rhs[2]) return false;
+
+        return true;
+    }
+
+    /// @notice Inequality comparator for two semantic version numbers.
+    /// @param lhs The left-hand side semantic version number.
+    /// @param lhs The right-hand side semantic version number.
+    /// @return Whether the two numbers are inequal or not.
+    function neq(uint8[3] memory lhs, uint8[3] memory rhs) internal pure returns (bool) {
+        if (lhs[0] != rhs[0]) return true;
+
+        if (lhs[1] != rhs[1]) return true;
+
+        if (lhs[2] != rhs[2]) return true;
+
+        return false;
+    }
```

Comment:
Why don't you use the `eq()` function and return the inverted result?

Reply:
This one is a bit cheaper + you don't need to include `eq` in your bytecode if you just need `neq`.

Reply:
true, but only if eq isn't used. up to you

Reply:
Then let's leave it as is